lint-hrx.ts

```diff
@@ -0,0 +1,161 @@
+import fs from 'fs';
+import p from 'path';
+import yargs from 'yargs/yargs';
+import fileLinesCount from 'file-lines-count';
+
+import {fromPath, SpecDirectory} from './lib-js/spec-directory';
+import RealDirectory from './lib-js/spec-directory/real-directory';
+import VirtualDirectory from './lib-js/spec-directory/virtual-directory';
+import {GithubActionsReporter} from './lib-js/reporter';
+
+/** The number of lines allowed in HRX files. */
+const LINE_THRESHOLD = 500;
+
+/**
+ * Verifies all HRX files in the project `spec` directory are under the
+ * {@link LINE_THRESHOLD}.
+ *
+ * HRX files are unwrapped and transformed into real files when
+ * {@link reporter.fix} is true.
+ *
+ * Skips HRX files if they include a top-level `options.yml` file annotated with
+ * :todo: or :ignore-for: `lint-hrx`. For example:
+ *
+ * ```hrx
+ * <===> options.yml
+ * # Ignored because <SOME REASON>
+ * :todo:
+ * - lint-hrx
+ * ```
+ */
+async function lintHrxSize(rootPath: string, reporter: GithubActionsReporter) {
+  const rootDir = await fromPath(rootPath);
+  const tooBig = await bigHrxInTree(rootDir, reporter);
+
+  // No Errors.
+  if (tooBig.length === 0) return;
+
+  if (!reporter.fix) {
+    const tooBigRelative = tooBig.map(f => p.relative(process.cwd(), f));
+    for (const f of tooBigRelative) {
+      reporter.reportError(`HRX file exceeds ${LINE_THRESHOLD} lines`, f, true);
+    }
+    return;
```

Comment:
Nit: since the `else` branch is approximately the same size as the `if` branch here, I'd just use an `else` explicitly rather than an early return.